### PR TITLE
[DNM][WIP] Changeling Containment Suit

### PR DESCRIPTION
**Pull request by Quantum-M**
**What does this PR do:**
This PR includes a Suit and a Helmet, called Null Suit and Null Helmet respectively, that nullifies most powers changelings have when **_both_** pieces are worn, excluding some passive powers.

The intent is to give Security, and changelings, a option in which the changeling doesn't have to worry about being captured and ending cremated/gibbed and stuck out of the round, which I believe is a factor as to why changelings tends to be murderous, especially towards Security, and it isn't a interesting when changelings ends up stuck out of the round

Powers that will not work (All aggressive powers and some passive ones):

- Shield
- Changeling blade
- Changeling Armor and Space Suit
- Both forms of shrieks
- Chameleon and digital tracking (may have to test to see if, when active, it is removed when the suit is placed on them)
- Last Resort (was unsure on this one)
- Hivemind link
- Bodyswap
- Absorb
- All forms of stings (changelings can activate it, but it will not be able to use against others)
- Lesser form (the monkey one)

Powers that still work:
- Hivemind talk
- Revive Stasis (Or whatever it's called)
- The fake voice power
- Anatomic Panacea
- Adrenaline glands

I may have missed some powers with this list, I will update them after I look over them again.

**Progress**
- [X] Get the suit to work on nullifying powers
- [ ] Toning/ adjust sprites Sprites
- [ ] Figuring out how to lock both the suit and helmet (have tried to look how the vampire muzzle worked and had some issues
- [ ] Figuring how to have the helmet not be dropped on revived (NODROP flag may help on this)
- [ ] Map some suits to Security, most likely two sets
- [ ] Make one set of suit be able to be ordered from cargo, albeit at expensive levels.

**NOTE!** Some suggestions on figuring out how to "lock" the suit would be appreciated, as I tried looking at it and had some issues and no success.

**Images of sprite/map changes (IF APPLICABLE):**
To be added.

**Changelog:**

:cl: Quantum-M
add: Adds a new Changeling Containment Helmet and Suit, both called Null Helmet and Null Suit respectively, which allows to a more safer containment of changelings.
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
I don't see who this hurts by including it but i'm almost 100% sure security would still just murder and gib changelings

**Comment by rb303** [comment]
I only see this being used for changelings who surrender, really. Wouldn't blame security for just killing changelings over lugging around a suit to capture them.

**Comment by taukausanake** [comment]
I don't know. This isn't bad per se, but I don't see it getting merged. I don't think the maints will care for it much

File: code/modules/clothing/suits/utility.dm [hunk 1]
```diff
@@ -156,3 +157,113 @@
...
+		"Tajaran" = 'icons/mob/species/tajaran/helmet.dmi',
+		"Vulpkanin" = 'icons/mob/species/vulpkanin/helmet.dmi',
+		"Grey" = 'icons/mob/species/grey/helmet.dmi',
+		"Drask" = 'icons/mob/species/drask/helmet.dmi',
+		"Vox" = 'icons/mob/species/vox/helmet.dmi',
+		"Skrell" = 'icons/mob/species/skrell/helmet.dmi'
+		)
+
+/obj/item/clothing/head/null_suit/attack_hand(mob/user as mob)
+	if(user:head == src && !user.IsAdvancedToolUser())
```
Comment (Fox-McCloud) [inline code/modules/clothing/suits/utility.dm]: I'm going to unironically stab you.

**Comment by Fox-McCloud** [comment]
Because spending the rest of the shift locked up in permabrig is so much more fun.

**Comment by Shadeykins** [comment]
I think this moves in the wrong direction.

If we want changelings to be less murderous, we should adjust them to make them less overtly powerful (and more situationally powerful). Changelings were originally intended to be agents of social infiltration in a Jack Carpenter's _The Thing_ style of fashion.

All this does is reward changelings who hand themselves in and play friendly with security, something that was a massive issue back when we had Loyalty Implants.

This allows security to create a "safe changeling" that can work with them and use hivemind to bait other changelings.

**Comment by TDSSS** [comment]
If security really wants to contain a changeling, can't they already do it with some shield generators? Just don't give them a vent or a window to space to break and there's not much they can do, right? It's not like vampires with misting.

**Comment by taukausanake** [comment]
Heh. Treat them like a xeno and stick them in Xenobiology. That would be interesting though.

My simple solution would just be a tracking implant, maybe one that had shielding to prevent EMPs or create an intense shock and stun if it is destroyed so security can run to the last known location and detain them (assuming no one dragged them away). Hell, that would be useful for any criminal let out on parole, make that instead

**Comment by SkeletalElite** [comment]
Totally against this. Changelings is they are supposed to be difficult to kill uncontainable monsters. Plus you can still technically contain a ling by locking it in solitary with the vents removed and walling the door so they cant break it with their arm.

**Comment by Shadeykins** [comment]
> 
> 
> If security really wants to contain a changeling, can't they already do it with some shield generators? Just don't give them a vent or a window to space to break and there's not much they can do, right? It's not like vampires with misting.

Nobody wants to be locked in a 2x2 box. Creating a shield generator just for 1 changeling is not tenable in most cases.

This allows security to safely outfit a changeling with a roundstart item in a way that allows the changeling to still manuever and wander around (which is great for the changeling player).

A 2x2 box that nobody visits is dramatically different from being allowed to roam around freely. Changelings are defacto KOS for good reason and it's to entirely prevent the headache that is friendly changelings working for security (and yes, this was a massive issue when they could be reliably 'contained' in the past).

File: code/game/gamemodes/changeling/changeling_power.dm [hunk 2]
```diff
@@ -51,6 +52,12 @@
 
 //Fairly important to remember to return 1 on success >.<
 /obj/effect/proc_holder/changeling/proc/can_sting(var/mob/user, var/mob/target)
+
+	if(nullified == 1) //If the power can be nullified by the containment suit, checks if the user is wearing the suit and helmet
+		if(istype(user:wear_suit, /obj/item/clothing/suit/null_suit) && istype(user:head, /obj/item/clothing/head/null_suit))
```
Comment (Citinited) [inline code/game/gamemodes/changeling/changeling_power.dm]: `user:wear_suit` 😱

**Comment by variableundefined** [comment]
I echo the concerns above.

**Comment by Quantum-M** [comment]
Lost the motivation to continue this particular PR. Anyone is free to use the codework and/or spritework I have made for the PR.

**Pull request closed without merging**